drivers/soundwire/generic_bandwidth_allocation.c

```diff
@@ -314,7 +314,7 @@ static int sdw_select_row_col(struct sdw_bus *bus, int clk_freq)
 			frame_freq = clk_freq / frame_int;
 
 			if ((clk_freq - (frame_freq * SDW_FRAME_CTRL_BITS)) <
-			    bus->params.bandwidth)
+			    bandwidth)
```

Comment:
I don't follow this patch at all.

First, we need to avoid that the frame size selected depends on the bandwidth required on each lane. It would be a disaster we had different results for each lane.

Second, this definition
````
include/linux/soundwire/sdw.h:#define SDW_FRAME_CTRL_BITS               48
````

is ONLY valid for lane0. There are no control bits on non-zero lanes, we can use the entire set of columns for audio data (or BPT/BRA).

Reply:
Let me think more about it. My intent is to allocate lane according to available bandwidth on each lane. That's why I extend bandwidth to an array.

Reply:
Hmm, I guess I can use a new lane if current bandwidth is not enough in sdw_compute_bus_params(). I will try it.